specs/marketplace.md

```diff
@@ -14,58 +14,52 @@ contributors:
+The marketplace is a critical component of the Codex network, serving as a platform where all involved parties interact to ensure data persistence. It also provides mechanisms to enforce agreements and facilitate data repair when Storage Nodes fail to fulfill their duties.
 
-The marketplace is defined by a smart contract deployed to an EVM-compatible blockchain. It has several flows which are 
-linked with roles in the network and which the participating node takes upon itself. Each node can take on the responsibilities of one role or multiple at the same time.
-This specification describes these flows.
+Implemented as a Smart Contract on an EVM-compatible blockchain, the Marketplace enables various scenarios where nodes assume one or more roles to maintain a reliable persistence layer for users. This specification details these interactions.
 
-The Marketplace handles storage requests, the storage slot state,
-storage provider rewards, storage provider collaterals, and storage proof state.
+The Marketplace contract manages Storage Requests, maintains the state of allocated Storage Slots, and orchestrates Storage Provider rewards, collaterals, and storage proofs.
 
-If a node implementation wants to participate in the persistence layer of Codex, it needs to choose which role(s) it wants
-to support and implement proper flows. Otherwise, it won't be compatible with the rest of the Codex network.
+A node that wishes to participate in the Codex persistence layer MUST implement one or more roles described in this document.
 
 ### Roles
 
-There are three main roles in the network: client, storage provider (SP) and validator. 
+A node can assume one of the three main roles in the network: the Client, the Storage Provider (SP), and the Validator (for the sake of conciseness and where it does not cause ambiguity, referred to as a Client, Storage Provider, and Validator, respectively).
 
-A client is a potentially short-lived node in the network that mainly interacts with the purpose of persisting
-its data in the network. 
+A Client is a potentially short-lived node in the network with the purpose of persisting its data in the Codex persistence layer.
 
-A Storage Provider is a long-term participant in the network that stores other data for profit. It needs to provide a proof
-to the smart contract that it possesses the data from time-to-time.
+A Storage Provider is a long-lived node providing storage for Clients in exchange for profit. To ensure a reliable, robust service for Clients, Storage Providers are required to periodically provide evidence that they are persisting the data.
 
-A validator validates that SPs are correctly fulfilling their duties and that they provide proofs of storage
-on time.
+A Validator ensures that Storage Providers correctly fulfill their duties and provide proofs of storage when requested by the Smart Contract.
```

Comment:
Yes, that how I understood it as well. My use of `and` in the sentence is confusing as it may indicate that it is the validator who provides proofs... There is also a comment from @Eric on it, which makes it clear.